src/app/calculator/calculator.component.ts

```diff
@@ -157,42 +264,170 @@ export class CalculatorComponent {
+    return coordKnee;
+  }
 
-    const yInMiddle = heightSaddlePixel / 2;
-    const xMedianCorner = Math.sqrt(Math.pow(waistToKneePixel, 2) - Math.pow(yInMiddle, 2));
+  calculateCoordinateFootOnGround(heightSaddlePixel: number) {
+    if (this.riderValues.legPixel > heightSaddlePixel) {
+      return {
+        x: this.getCoordinatesCenterSaddleX(),
+        y: this.getCoordinatesCenterSaddleY() + heightSaddlePixel,
+      };
+    }
+    return {
+      x: this.getCoordinatesCenterSaddleX(),
+      y: this.getCoordinatesCenterSaddleY() + this.riderValues.legPixel,
+    };
+  }
 
-    if (legPixel > heightSaddlePixel) {
-      this.canvasLeg.passValue({
-        xSaddle, ySaddle, xMedianCorner, yInMiddle, heightSaddlePixel, footPixel, isCorner: true,
-      });
+  calculateLengthBetweenTwoPoints = (A, B: CoordinateModel) => {
+    if (A === null || B === null) {
+      return null;
     }
-    if (legPixel <= heightSaddlePixel) {
-      this.canvasLeg.passValue({
-        xSaddle, ySaddle, legPixel, footPixel, isCorner: false,
-      });
+    return Math.sqrt(Math.pow((B.x - A.x), 2) + Math.pow(B.y - A.y, 2));
+  }
+
+  calculateCoordinateThirdCornerOfTriangle(coordinateFirstCorner: CoordinateModel,
+    coordinateSecondCorner: CoordinateModel, lenghtFirstSide, lenghtSecondSide: number) {
+    if (
+      coordinateFirstCorner === null
+      || coordinateSecondCorner === null
+      || lenghtFirstSide === null
+      || lenghtSecondSide === null
+    ) {
+      return null;
     }
+    const lenghtThirdSide = this.calculateLengthBetweenTwoPoints(coordinateFirstCorner,
+      coordinateSecondCorner);
+    const segmentThirdSideDividedByHeight = (
+      Math.pow(lenghtFirstSide, 2)
+      - Math.pow(lenghtSecondSide, 2)
+      + Math.pow(lenghtThirdSide, 2)
+    )
+      / (lenghtThirdSide * 2);
+    const heightOfThirdCorner = Math.sqrt(
+      Math.pow(lenghtFirstSide, 2)
+      - Math.pow(segmentThirdSideDividedByHeight, 2),
+    );
+    const pointThirdSideDividedByHeight = { x: null, y: null };
+    pointThirdSideDividedByHeight.x = coordinateFirstCorner.x
+      + (
+        (segmentThirdSideDividedByHeight / lenghtThirdSide)
+        * (coordinateSecondCorner.x - coordinateFirstCorner.x)
+      );
+    pointThirdSideDividedByHeight.y = coordinateFirstCorner.y
+      + (
+        (segmentThirdSideDividedByHeight / lenghtThirdSide)
+        * (coordinateSecondCorner.y - coordinateFirstCorner.y)
+      );
+    const coordinateThirdCorner = { x: null, y: null };
+    coordinateThirdCorner.x = pointThirdSideDividedByHeight.x
+      + (
+        (heightOfThirdCorner / lenghtThirdSide)
+        * (coordinateSecondCorner.y - coordinateFirstCorner.y)
+      );
+    coordinateThirdCorner.y = pointThirdSideDividedByHeight.y
+      - (
+        (heightOfThirdCorner / lenghtThirdSide)
+        * (coordinateSecondCorner.x - coordinateFirstCorner.x)
+      );
+    return coordinateThirdCorner;
+  }
+
+  showRider() {
+    this.calculateRiderValues();
```

Comment:
I suppose you should think about pure functions. Like calculateRiders should return object with coordinates, and showLegs, should receive coordinates and draw. This function shouldn't know nothing about data outside it.

Reply:
ok. I'll try to change